dotnet/src/AutoGen.Anthropic/Middleware/AnthropicMessageConnector.cs

```diff
@@ -85,20 +85,46 @@ private async Task<IEnumerable<IMessage>> ProcessMessageAsync(IEnumerable<IMessa
 
         return processedMessages;
     }
-
     private IMessage PostProcessMessage(ChatCompletionResponse response, IAgent from)
     {
         if (response.Content is null)
         {
             throw new ArgumentNullException(nameof(response.Content));
         }
 
-        if (response.Content.Count != 1)
+        // When expecting a tool call, sometimes the response will contain two messages, one chat and one tool.
```

Comment:
This middleware also needs to process the following input message types to work with other function call agents
- ToolCallMessage
- ToolCallResultMessage
- ToolCallAggregateMessage

You can refer to `OpenAI` or `Gemini` on how those messages being converted

Reply:
Could you give me feedback on this ? i followed the pattern of `OpenAi` and `Gemini`. Wrote Ut's that invoke these methods and they all pass. Looking for second opinion if i did this correctly. Hopefully the changes do not hurt your eyes 😁

Reply:
Sorry for the late reply, was afk in the past few days,,

These message types (`ToolCallMessage`, `ToolCallResultMessage` and `ToolCallAggregateMessage`) also need to be processed so that the agent can process function call message types correctly when they are in the chat history.

For example, considering the following message history
```
TextMessage("What's the weather in seattle", from: user)
ToolCallAggregateMessage(ToolCallMessage("getWeather"), ToolCallResultMessage("the weather in seattle is sunny"))
```

To perform a multi-turn function call where the function call result needs to be sent to LLM for further response, the agent also needs to support `ToolCallAggregateMessage` as input.

Similiary, `ToolCallMessage` and `ToolCallResultMessage` needs to be supported as well.

Reply:
sorry for the delay, been a little busy with regular day job  😆
`
I added support for `ToolCallMessage`, `ToolCallAggregateMessage`, and `ToolCallResultMessage` 

Let me know if this looks good to you, added unit tests as well.